### ISSUE TITLE
Support for CSS Guidelines

### ISSUE THREAD

**Issue by psren** [geddski/csstyle#33]
Would you accept a Pull-Request for a config switch to support http://cssguidelin.es/ ?

```scss
$csstyle-support-guidelines: true;
@import 'csstyle';
```

Then the ```options``` can be used like
```html
<div class="component component--option">
```

instead of
```html
<div class="component --option">
```

**Comment by geddski** [comment]
Yeah I know a couple people that would like this. Though I'd probably use a different config name since css guidelines includes a lot of other things than this. 



—

**Comment by psren** [comment]
Do you have any idea about a better config name?
```csstyle-long-options``` ?

**Comment by benseitz** [comment]
:+1:  I would love that.

**Comment by psren** [comment]
Ping @geddski :-)
I'm preparing a PR for that. If you have a better suggestiion for the config-switch let me know.

**Comment by geddski** [comment]
Could go with $csstyle-verbose-options or $csstyle-long-options



—

**Comment by teddybradford** [comment]
Any news on this front? Would love to use an option like this.

**Comment by psren** [comment]
Sorry for the delay. i must have missed your answer regarding the variable namings.

**Comment by psren** [comment]
I in general think that the ```.component--option``` is a bit better for the following reason

```html
<div class="person --male"></div>
<div class="monkey --male"></div
```

```--male``` is not unique so one style affects each other.
And the specitifity is a little lower with the guildline.es way.

**Comment by teddybradford** [comment]
Agreed. The `.component--option` naming convention also keeps components more isolated, which is especially helpful when debugging.

**Comment by geddski** [comment]
As long as you're using csstyle to create your selectors the options will never collide because it includes both the component and the option name in the selector.

**Comment by geddski** [comment]
The main win of this PR is to make grepping for selectors in your html slightly easier.

**Comment by teddybradford** [comment]
True, but I can also see a use case for wanting to use csstyle with an already existing codebase that has an established library of components. Depending on the scale and complexity of the project, refactoring all components to use csstyle may not be practical. With that said, you're probably right that easier grepping is the main allure for this.

**Comment by geddski** [comment]
Good point. This makes csstyle that much easier to configure to an existing convention like many big projects would already have. +1

@psren Are you ready for me to review and merge?

**Comment by psren** [comment]
Des. Would be nice if you could look at the missing postcss test.

**Comment by geddski** [comment]
Will do

**Comment by psren** [comment]
Is there any news here?

**Comment by psren** [comment]
Ping

**Comment by psren** [comment]
Pong

**Comment by SimonJ** [comment]
ding dong
